### ISSUE TITLE
About comparasion with traditional Info-NCE loss(with positive sample)

### ISSUE THREAD

**Issue by kmp1001** [raywang4/DispLoss#9]
Hello, thank you for your great work! I want to implement experiment that uses a traditional Info-NCE loss(with positive sample).   
I try to implement like this,
```python
def info_nce_l2_loss(z, z1, temperature=1.0):
    dual_z = torch.cat([z, z1], dim=0)
    labels = torch.cat([torch.arange(z.shape[0]) for _ in range(2)], dim=0)
    labels = (labels.unsqueeze(0) == labels.unsqueeze(1)).float()

    dual_z = dual_z.reshape(dual_z.shape[0], -1)
    dist2 = torch.cdist(dual_z, dual_z, p=2).pow(2)
    dist2 = dist2 / dual_z.shape[1]
    similarity_matrix = -dist2

    mask = torch.eye(labels.shape[0], dtype=torch.bool, device=dual_z.device)
    labels = labels[~mask].view(labels.shape[0], -1)
    similarity_matrix = similarity_matrix[~mask].view(similarity_matrix.shape[0], -1)

    positives = similarity_matrix[labels.bool()].view(labels.shape[0], -1)
    negatives = similarity_matrix[~labels.bool()].view(similarity_matrix.shape[0], -1)
    logits = torch.cat([positives, negatives], dim=1) / temperature

    labels_ce = torch.zeros(logits.shape[0], dtype=torch.long, device=dual_z.device)
    return F.cross_entropy(logits, labels_ce)
```

However, it seems that info_nce_l2 loss would not work after 1 epoch, specifically, flow matching loss and info_nce loss is nan. Have you ever met this strange issue? Or could you please provide with this experiment's code? THKs!

**Comment by raywang4** [comment]
Hi there, thanks for the interest in our work! In our experience, NaN issues are usually caused by division over zero during loss calculation. You might want to check (especially for torch builtin functions like cdist or cross entropy) whether that happens in your code. If this is the case, adding a small constant like 1e-8 should fix the problem.
